src/game_config.cpp

```diff
@@ -260,6 +275,129 @@ Filesystem_Stream::OutputStream Game_Config::GetGlobalConfigFileOutput() {
 	return Filesystem_Stream::OutputStream();
 }
 
+Filesystem_Stream::OutputStream& Game_Config::GetLogFileOutput() {
+	if (!Player::player_config.log_enabled.Get()) {
+		static Filesystem_Stream::OutputStream noop_stream;
+		return noop_stream;
+	}
+
+	if (!logging.started) {
+		logging.started = true;
+
+		std::string path;
+
+		if (logging.path.empty()) {
+	#if defined(_WIN32)
+			PWSTR knownPath;
+			const auto hresult = SHGetKnownFolderPath(FOLDERID_LocalAppData, 0, nullptr, &knownPath);
+			if (SUCCEEDED(hresult)) {
+				path = Utils::FromWideString(knownPath);
+				CoTaskMemFree(knownPath);
+			} else {
+				Output::Debug("LogFile: SHGetKnownFolderPath failed");
+			}
+	#elif defined(SYSTEM_DESKTOP_LINUX_BSD_MACOS)
+			char* home = getenv("XDG_STATE_HOME");
+			if (home) {
+				path = home;
+			} else {
+				home = getenv("HOME");
+				if (home) {
+					path = FileFinder::MakePath(home, ".local/state");
+				}
+			}
+
+			if (!path.empty()) {
+				path = FileFinder::MakePath(path, OUTPUT_FILENAME);
+			}
```

Comment:
missing "if empty" here?

This could potentially remove the `#else` case as well.